application/controllers/QuestionAdministrationController.php

```diff
@@ -2852,54 +2885,50 @@ private function updateSubquestions($question, $subquestionsArray)
     {
         $questionOrder = 0;
         foreach ($subquestionsArray as $subquestionId => $subquestionArray) {
-            foreach ($subquestionArray as $scaleId => $data) {
-                $subquestion = Question::model()->findByAttributes(
-                    [
-                        'parent_qid' => $question->qid,
-                        'title'      => $data['code'],
-                        'scale_id'   => $scaleId
-                    ]
-                );
-                if (empty($subquestion)) {
-                    throw new Exception('Found no subquestion with code ' . $data['code']);
-                }
-                $subquestion->sid        = $question->sid;
-                $subquestion->gid        = $question->gid;
-                $subquestion->parent_qid = $question->qid;
-                $subquestion->question_order = $questionOrder;
-                $questionOrder++;
-                if (!isset($data['code'])) {
-                    throw new CHttpException(
-                        500,
-                        'Internal error: Missing mandatory field code for question: ' . json_encode($data)
-                    );
-                }
-                $subquestion->title      = $data['code'];
-                if ($scaleId === 0) {
-                    $subquestion->relevance  = $data['relevance'];
-                }
-                $subquestion->scale_id   = $scaleId;
-                if (!$subquestion->update()) {
-                    throw (new LSUserException(500, gT("Could not save subquestion")))
-                        ->setDetailedErrorsFromModel($subquestion);
-                }
-                $subquestion->refresh();
-                foreach ($data['subquestionl10n'] as $lang => $questionText) {
-                    $l10n = QuestionL10n::model()->findByAttributes(
-                        [
-                            'qid' => $subquestion->qid,
-                            'language' => $lang
-                        ]
-                    );
-                    if (empty($l10n)) {
-                        $l10n = new QuestionL10n();
+            if (substr($subquestionId, 0, 3) !== "new") {
+                foreach ($subquestionArray as $scaleId => $data) {
+                    $subquestion = Question::model()->findByPk($subquestionId);
+                    if (empty($subquestion)) {
+                        throw new Exception('Found no subquestion with code ' . $subquestionId);
```

Comment:
`Found no subquestion with id`

BUT : i don't think we must throw an error , maybe add a flashMessage + create the sub-question ?

Reply:
PS : must not happen